### PR TITLE
refactor/cdn use product offers

### PR DESCRIPTION
**Pull request by JeremyDec**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | yes
| Tickets          | Fix #...
| License          | BSD 3-Clause

## Description

This work is made before the feature for Shared CDN Security-Advanced feature

Use of the `ovh-manager-product-offers` component into CDN order/activate/upgrade part, in Web Cloud manager.
A fix has also been done to go to next step automatically, when using services workflow.

The services (using /services routes) has been updated (with its dedicated service `product-offers-services`) to be used also for /services/*/upgrade routes.

:warning: **BREAKING CHANGE**:  (See df91d77b5cc274d1e43e8c974d8ef3e2c5022f6e)

1.  name of the property `detachPlancodes` has been renamed to `plancodes`. 
2. name of the services workflow validate result is renamed `result` (was `detachResult` before)

This PR contains also the required change to adapt to the module breaking change. (See e34356ef8058433f2ecb126797cd420e01ebf793)

ℹ️  For Web cloud section, **small refactor** has been done to just use the component. Aside of the refactor, the code has not changed.

### CODE REVIEW

File: packages/manager/modules/product-offers/src/workflows/product-offers-services-workflow.class.js [hunk 1]
```diff
@@ -32,24 +32,27 @@ export default class ServicesWorkflow extends Workflow {
...
-    this.pricings = this.computePricing(detachPlancode.prices);
+    const [actionPlancode] = this.plancodes; // Yes, for the moment, there is just one plan code
+    this.plancode = actionPlancode.planCode;
+    this.pricings = this.computePricing(actionPlancode.prices);
 
     if (this.hasUniquePricing()) {
       this.$timeout(() => {
         this.currentIndex += 1;
         [this.pricing] = this.pricings;
+        if (typeof this.onPricingSubmit === 'function') {
```
Comment (zakariasahmane) [inline packages/manager/modules/product-offers/src/workflows/product-offers-services-workflow.class.js:47]: Is this verification necessary? Based on the name "onPricingSubmit" it sounds like it'll always be a function unless it's null or undefined, no?

**Review by zakariasahmane — APPROVED** [review]

File: packages/manager/modules/product-offers/src/workflows/product-offers-services-workflow.class.js (hunk 1 above)
Comment (JeremyDec) [reply packages/manager/modules/product-offers/src/workflows/product-offers-services-workflow.class.js:47]: It is to test also if `onSubmitFunction` is a function and not an object. But you're right its name clearly indicates it must be a function. It's mainly to replace `this.onPricingSubmit && isFunction(onPricingSubmit)` kind of test we usually used to do with Lodash.

File: packages/manager/modules/product-offers/README.md [hunk 2]
```diff
@@ -71,6 +71,8 @@ export default class {
       getPlanCode: this.getPlanCode.bind(this),
       productName: 'webHosting',
       serviceNameToAddProduct: 'nameOfMyProduct',
+      onPricingSubmit: () => console.log('onPricingSubmit'),
+      onValidateSubmit: () => console.log('onValidateSubmit'),
```
Comment (radireddy) [inline packages/manager/modules/product-offers/README.md:75]:
```suggestion

```

**Comment by radireddy** [comment]
@JeremyDec "Product-offers" component is used in 'Platform.sh'. This is not yet merged to develop/master. Tomorrow is the prod date. Does your change has any impact on it? I see you made some changes to this component and also refactored the code.

File: packages/manager/modules/product-offers/README.md (hunk 2 above)
Comment (JeremyDec) [reply packages/manager/modules/product-offers/README.md:75]: It is in the README file ;), to display the possibility to use those two methods

**Comment by JeremyDec** [comment]
Rebase on `develop` to resolve conflict. 1 approval before (from @zakariasahmane)

**Comment by JeremyDec** [comment]
@radireddy, I've look into your Platform SH, you're using the `order` type of workflow, so it won't be broken by these changes. Also I can wait after the prod to merge the PR

**Review by frenautvh — DISMISSED** [review]

**Comment by JeremyDec** [comment]
Rebase due to conflict, had 2 approvals before

File: packages/manager/apps/web/client/app/hosting/cdn/order/hosting-cdn-order.component.js [hunk 3]
```diff
@@ -1,31 +1,17 @@
...
     hasCDN: '<',
-    isV1CDN: '<',
-    isIncludedCDN: '<',
-    isPayableCDN: '<',
-    isV2CDN: '<',
     cdnCase: '<',
-    trackClick: '<',
+
+    onError: '<',
+    onSuccess: '<',
```
Comment (radireddy) [inline packages/manager/apps/web/client/app/hosting/cdn/order/hosting-cdn-order.component.js:12]: do you think & will be more appropriority here?

**Review by radireddy — APPROVED** [review]

**Pull request merged**